File: Front/TS/src/infrastructure/adapters/repository/InMemoryUserRepository.ts
import { IMemberGateway, MemberInfo } from '../../../domain/abstractions/IMemberGateway';
import { IUserContext } from '../../../domain/abstractions/IUserContext';
import { IUserGateway } from '../../../domain/abstractions/IUserGateway';
export class InMemoryUserRepository implements IUserGateway, IMemberGateway {
constructor(private readonly userContext: IUserContext, public readonly users: any[] = []){}
    RetrieveInfo(): Promise<MemberInfo> {
        const member = this.users.find(x => x.id == this.userContext.UserId);
        return Promise.resolve({
            coins: 2000,
            username: member.username
        })
    }

    Register(user: { id: string; username: string; email: string; password: string; }): Promise<void> {
        this.users.push(user);
        this.userContext.UserId = user.id
        return Promise.resolve();
    }

}

File: Back/.Net/Modules/Users/BetFriends.Users.Infrastructure/Repositories/InMemoryUserRepository.cs
using BetFriends.Shared.Infrastructure.Event;
using BetFriends.Users.Domain.Users;

namespace BetFriends.Users.Infrastructure.Repositories;

internal class InMemoryUserRepository(DomainEventsAccessor domainEventsAccessor) : IUserRepository
{
    private readonly List<User> users = [];
    public Task<bool> IsUserExistAsync(string username, string email)
    {
        if(users.Any(x => x.Snapshot.Username == username || x.Snapshot.Email == email))
            return Task.FromResult(true);
        return Task.FromResult(false);
    }

    public Task SaveAsync(User user)
    {
        users.Add(user);
        domainEventsAccessor.Add(user.Events);
        return Task.CompletedTask;
    }

    internal User GetUser(string email, string password)
    {
        return users.Find(x => x.Snapshot.Email == email && x.Snapshot.Password == password)!;
    }
}
